QuantLib/ql/termstructures/yield/fittedbonddiscountcurve.cpp

```diff
@@ -250,8 +275,37 @@ namespace QuantLib {
     Disposable<Array>
     FittedBondDiscountCurve::FittingMethod::FittingCost::values(
                                                        const Array &x) const {
-        Array y(1, value(x));
-        return y;
+        Date refDate  = fittingMethod_->curve_->referenceDate();
```

Comment:
This is ok, but I'd avoid the code duplication in `value` and `values`. You can implement `value(x)` by calling `values(x)` and adding the errors in the returned vector.